lib/connectors/memory.js

```diff
@@ -421,6 +421,11 @@ function applyFilter(filter) {
       // Also support array types. Mongo, possibly PostgreSQL
       if (Array.isArray(value)) {
         var matcher  = where[key];
+        // The following condition is for the case where we are querying with
+        // a neq filter, and when the value is an empty array ([]).
+        if (matcher.neq !== undefined && value.length <= 0) {
```

Comment:
For a value, I interpret {neq: [1, 2]} as not equal to 1 or 2. 3 matches, 1 does not match, 2 does not match.
For an array, I would interpret it as does not contain 1 or 2. [3,4,5] matches, [1,3,4,5] does not match, [2,3,4,5] does not match.
Anyway, is this syntax should really be supported by loopback ? I can't find it in the documentation ...